### ISSUE TITLE
[Bug] Reducing with 1D tensor and dim=0 gives FATAL | Output tensor size mismatch

### ISSUE THREAD

**Issue by pglusacTT** [tenstorrent/tt-forge-fe#907]
# Description

Doing ReduceAvg(..., dim=-1) and not with ReduceAvg(..., dim=0) on a 1D tensor results in an error. 

# Reproduce

Run the following code

```python
import torch
import forge


class Module(forge.ForgeModule):
    def __init__(self, name="module"):
        super().__init__(name)

    def forward(self, a):
        red = forge.op.ReduceAvg("red", a, dim=0)
        return red

a = torch.tensor([1, 2, 3]).float()
af = forge.tensor.Tensor.create_from_torch(a)

uncomp_module = Module()
module = forge.compile(uncomp_module, sample_inputs=[af])

output = module(af)
print(output)
```

Error message: FATAL | Output tensor size mismatch
